manifests.go

```diff
@@ -0,0 +1,66 @@
+package distribution
+
+import (
+	"github.com/docker/distribution/context"
+	"github.com/docker/distribution/digest"
+	"github.com/docker/distribution/reference"
+)
+
+// Manifest represents a registry object specifying a target and a set of constituents
+type Manifest interface {
+	// Target returns a descriptor for the configuration of this manifest.  This
+	// may return a nil descriptor if none exists for this manifest.
+	Target() Descriptor
```

Comment:
I did consider this, but didn't want to fragment the interface.

Perhaps more explicit return types (e.g. returning ErrNotSupported) would work here.